gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+    bins_x : `~numpy.array` of `~astropy.coordinates.Angle`
+        array with the bin edges for the X coordinate
+    bins_y : `~numpy.array` of `~astropy.coordinates.Angle`
+        array with the bin edges for the Y coordinate
+
+    Returns
+    -------
+    w : `~astropy.wcs.WCS`
+        WCS object describing the bin coordinates
+    """
+    # check units
+    unit_x = bins_x.unit
+    unit_y = bins_y.unit
+    if unit_x != unit_y:
+        ss_error = "Units of X ({0}) and Y ({1}) bins do not match!".format(
+            unit_x, unit_y)
+        ss_error += " Is this expected?"
+        raise ValueError(ss_error)
+
+    # Create a new WCS object. The number of axes must be set from the start
+    w = wcs.WCS(naxis=2)
+
+    # Set up DET coordinates in degrees
+    nbins_x = len(bins_x) - 1
+    nbins_y = len(bins_y) - 1
+    range_x = Angle([bins_x[0], bins_x[-1]])
+    range_y = Angle([bins_y[0], bins_y[-1]])
+    delta_x = (range_x[1] - range_x[0])/nbins_x
+    delta_y = (range_y[1] - range_y[0])/nbins_y
+    w.wcs.ctype = [name_x, name_y]
+    w.wcs.cunit = [unit_x, unit_y]
+    w.wcs.cdelt = [delta_x.to(unit_x).value, delta_y.to(unit_y).value]
+    # ref as lower left corner (start of (X, Y) bin coordinates)
+    # coordinate start empiricaly determined at pix = 0.5: why 0.5?
+    w.wcs.crpix = [0.5, 0.5]
+    w.wcs.crval = [(bins_x[0] + (w.wcs.crpix[0] - 0.5)*delta_x).to(unit_x).value,
+                   (bins_y[0] + (w.wcs.crpix[1] - 0.5)*delta_y).to(unit_y).value]
+
+    return w
+
+
+class CubeBackgroundModel(object):
+
+    """Cube background model.
+
+    Container class for cube background model (X, Y, energy).
+    (X, Y) are detector coordinates (a.k.a. nominal system).
+    The class hass methods for reading a model from a fits file,
+    write a model to a fits file and plot the models.
+
+    Parameters
+    ----------
+    detx_bins : `~astropy.coordinates.Angle`
+        Spatial bin edges vector (low and high). X coordinate.
+    dety_bins : `~astropy.coordinates.Angle`
+        Spatial bin edges vector (low and high). Y coordinate.
+    energy_bins : `~astropy.units.Quantity`
+        Energy bin edges vector (low and high).
+    background : `~astropy.units.Quantity`
+        Background cube in (energy, X, Y) format.
+    """
+
+    def __init__(self, detx_bins, dety_bins, energy_bins, background):
+        self.detx_bins = detx_bins
+        self.dety_bins = dety_bins
+        self.energy_bins = energy_bins
+
+        self.background = background
+
+    @staticmethod
+    def from_fits_bin_table(tbhdu):
+        """Read cube background model from a fits binary table.
+
+        Parameters
+        ----------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the bg cube
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        hdu = tbhdu['BACKGROUND']
```

Comment:
This looks to me like you're actually passing in an `HDUList`, not a `BinTableHDU` like you say in the docstring.
Everything is in this one HDU, right?
If so, please stick with taking a `BinTableHDU` as input for this method, and remove this line / adapt the callers.